metronome/hotstuff/service/test/src/io/iohk/metronome/hotstuff/service/storage/ViewStateStorageProps.scala

```diff
@@ -0,0 +1,193 @@
+    type SKey  = Nothing
+  }
+  type TestAggreement = TestAggreement.type
+
+  type Namespace = String
+
+  object TestKVStoreState extends KVStoreState[Namespace]
+
+  type TestViewStateStorage = ViewStateStorage[Namespace, TestAggreement]
+
+  class StorageWrapper(
+      viewStateStorage: TestViewStateStorage,
+      private var store: TestKVStoreState.Store
+  ) {
+    def getStore = store
+
+    def write(
+        f: TestViewStateStorage => KVStore[Namespace, Unit]
+    ): Unit = {
+      store = TestKVStoreState.compile(f(viewStateStorage)).runS(store).value
+    }
+
+    def read[A](
+        f: TestViewStateStorage => KVStoreRead[Namespace, A]
+    ): A = {
+      TestKVStoreState.compile(f(viewStateStorage)).run(store)
+    }
+  }
+
+  type State = ViewStateStorage.Bundle[TestAggreement]
+  type Sut   = StorageWrapper
+
+  val genesisState = ViewStateStorage.Bundle
+    .fromGenesisQC[TestAggreement] {
+      QuorumCertificate[TestAggreement](
+        Phase.Prepare,
+        ViewNumber(1),
+        "",
+        GroupSignature(Nil)
+      )
+    }
+
+  /** The in-memory KVStoreState doesn't invoke the codecs. */
+  implicit def neverUsedCodec[T] =
+    Codec[T](
+      (_: T) => sys.error("Didn't expect to encode."),
+      (_: BitVector) => sys.error("Didn't expect to decode.")
+    )
+
+  @nowarn
+  override def canCreateNewSut(
+      newState: State,
+      initSuts: Traversable[State],
+      runningSuts: Traversable[Sut]
+  ): Boolean = true
+
+  override def initialPreCondition(state: State): Boolean =
+    state == genesisState
+
+  override def newSut(state: State): Sut = {
+    val init = TestKVStoreState.compile(
+      ViewStateStorage[Namespace, TestAggreement]("test-namespace", state)
+    )
+    val (store, storage) = init.run(Map.empty).value
+    new StorageWrapper(storage, store)
+  }
+
+  override def destroySut(sut: Sut): Unit = ()
+
+  override def genInitialState: Gen[State] = Gen.const(genesisState)
+
+  override def genCommand(state: State): Gen[Command] =
+    Gen.oneOf(
+      genSetViewNumber(state),
+      genSetQuorumCertificate(state),
+      genSetLastExecutedBlockHash(state),
+      genGetBundle
+    )
+
+  def genSetViewNumber(state: State) =
+    for {
+      d <- Gen.posNum[Long]
+      vn = ViewNumber(state.viewNumber + d)
+    } yield SetViewNumberCommand(vn)
+
+  def genSetQuorumCertificate(state: State) =
+    for {
+      p <- Gen.oneOf(Phase.Prepare, Phase.PreCommit, Phase.Commit)
+      h <- arbitrary[TestAggreement.Hash]
+      s <- arbitrary[TestAggreement.GSig]
+      qc = QuorumCertificate[TestAggreement](
+        p,
+        state.viewNumber,
+        h,
+        GroupSignature(s)
+      )
+    } yield SetQuorumCertificateCommand(qc)
+
+  def genSetLastExecutedBlockHash(state: State) =
+    for {
+      h <- Gen.oneOf(
+        state.prepareQC.blockHash,
+        state.lockedQC.blockHash,
+        state.commitQC.blockHash
+      )
+    } yield SetLastExecutedBlockHashCommand(h)
```

Comment:
```suggestion
  def genSetLastExecutedBlockHash(state: State) =
    Gen.oneOf(
        state.prepareQC.blockHash,
        state.lockedQC.blockHash,
        state.commitQC.blockHash
      ).map(SetLastExecutedBlockHashCommand(_))
```

Reply:
wouldn't this save a few lines? ;)

Reply:
Indeed! I think I just wanted to match the visual style of the other two generators above it which `yield` a command.

Reply:
Annoyingly `scalafmt` decided that we cannot go from 7 to 5 lines :disappointed: 
```scala
    Gen
      .oneOf(
        state.prepareQC.blockHash,
        state.lockedQC.blockHash,
        state.commitQC.blockHash
      )
      .map(SetLastExecutedBlockHashCommand(_))
```